### ISSUE TITLE
Issue when volume is deleted but not eradicate (pgroup not assigned / fact empty)

### ISSUE THREAD

**Issue by smutel** [Pure-Storage-Ansible/FlashArray-Collection#282]
**Describe the bug**
I would like to create a volume in a protection group and it does not work as expected when the same volume is deleted but not eradicate:
- The volume is created but the volume is not attached to the protection group
- The fact returned (volume array) is empty

**To Reproduce**
1. Create a playbook and execute it:
```yml
- name: Test
  hosts: localhost
  connection: local
  gather_facts: false
  become: no
  tasks:
    - name: Fact
      set_fact:
        ansible_python_interpreter: "/usr/bin/env python3"

    - name: Ensure volumes are configured in Purity
        purestorage.flasharray.purefa_volume:
        fa_url: "xxxxxx"                                                    
        api_token: "xxxxxx"                                           
        name: "test"                                                                                
        size: "1G"                                                                                  
        state: "present"
      register: pure_volumes_info

    - name: Debug
      debug:
        var: pure_volumes_info
```
2. Result:
```
PLAY [Test]
***************************************************

TASK [Fact]
***************************************************
ok: [localhost]

TASK [Ensure volumes are configured in Purity]
***************************************************
changed: [localhost]

TASK [Debug]
***************************************************
ok: [localhost] => 
  pure_volumes_info:
    changed: true
    failed: false
    volume:
      created: '2022-04-29T06:57:57Z'
      name: test
      nvme_nguid: ...
      page83_naa: ...
      promotion_status: promoted
      requested_promotion_state: promoted
      serial: ...
      size: 1073741824
      source: null
```
4. Destroy the volume in the flasharray (do not eradicate it)
5. Execute the playbook again
6. The volume fact returned by the task is empty:
```
TASK [Debug]
***************************************************
ok: [localhost] => 
  pure_volumes_info:
    changed: true
    failed: false
    volume: []
```
8. Destroy the volume in the flasharray (do not eradicate it)
9. Add the option pgroup to the volume task
10. Execute the playbook again
11. The volume is restored but not attached to the pgroup

**Expected behavior**
I would like to have the volume fact returned by the task not empty even if the volume is restored.
I would like to be able to change settings of a restored volume

**Environment:**
 - FlashArray C with Purity//FA 6.1.14
 - Ansible 2.11.11
 - FlashArray-Collection master
 - py-pure-client 1.23.0

**Comment by sdodsley** [comment]
The fix is included in Collection 1.13.0

**Comment by sdodsley** [comment]
Samual

The docs dso state that `pgroup` can only be used when creating a new
volume.

As this volume does already exist, although in a deleted state, the
`pgroup` parameter is ignored.

The best thing to do would be a to restore the volume and then use the
`purefa_pg to add the volume into the required protection group.

| *Simon Dodsley DD MBCS CITP* |  Director, Telco & Open Source Integration
Solutions
| Pure Storage, Inc. | +1 (510) 516-2682
| ***@***.*** | www.purestorage.com

*BOOK TIME WITH ME*: 30 minutes
<https://purestorage.chilipiper.com/book/me/simon-dodsley?type=my-meeting>

[image: http://www.purestorage.com/_media/header_logo.png?130529]
